### PR TITLE
Update to Ubuntu 22.04 and GCC 12.2

### PR DESCRIPTION
**Pull request by rleh**

### CODE REVIEW

File: avr.Dockerfile [hunk 1]
```diff
@@ -3,6 +3,6 @@ LABEL maintainer="Niklas Hauser <niklas.hauser@rwth-aachen.de>, Raphael Lehmann
 LABEL Description="Image for building and debugging modm for AVR"
 LABEL org.opencontainers.image.source https://github.com/modm-ext/docker-modm-build
 
-RUN wget -qO- https://github.com/modm-io/avr-gcc/releases/download/v11.2.0/modm-avr-gcc.tar.bz2 | tar xj -C /opt
+RUN wget -qO- https://github.com/modm-io/avr-gcc/releases/download/v12.1.0/modm-avr-gcc.tar.bz2 | tar xj -C /opt
```
Comment (salkinium) [inline avr.Dockerfile]: This doesn't exist yet?

File: cortex-m.Dockerfile [hunk 2]
```diff
@@ -3,6 +3,6 @@ LABEL maintainer="Niklas Hauser <niklas.hauser@rwth-aachen.de>, Raphael Lehmann
 LABEL Description="Image for building and debugging modm for ARM Cortex-M"
 LABEL org.opencontainers.image.source https://github.com/modm-ext/docker-modm-build
 
-RUN wget -qO- https://developer.arm.com/-/media/Files/downloads/gnu-rm/10.3-2021.10/gcc-arm-none-eabi-10.3-2021.10-x86_64-linux.tar.bz2 | tar -xj -C /opt
+RUN wget -qO- https://developer.arm.com/-/media/Files/downloads/gnu/12.2.rel1/binrel/arm-gnu-toolchain-12.2.rel1-x86_64-arm-none-eabi.tar.xz | tar -xJ -C /opt
```
Comment (salkinium) [inline cortex-m.Dockerfile:6]: Hot off the press, 22nd Dec!

File: avr.Dockerfile [hunk 3]
```diff
@@ -3,6 +3,6 @@ LABEL maintainer="Niklas Hauser <niklas.hauser@rwth-aachen.de>, Raphael Lehmann
 LABEL Description="Image for building and debugging modm for AVR"
 LABEL org.opencontainers.image.source https://github.com/modm-ext/docker-modm-build
 
-RUN wget -qO- https://github.com/modm-io/avr-gcc/releases/download/v11.2.0/modm-avr-gcc.tar.bz2 | tar xj -C /opt
+RUN wget -qO- https://github.com/modm-io/avr-gcc/releases/download/v12.2.0/modm-avr-gcc.tar.bz2 | tar xj -C /opt
```
Comment (salkinium) [inline avr.Dockerfile:6]: This doesn't exist yet? I'll see if I can make it appear.

Comment (salkinium) [reply avr.Dockerfile:6]: It has magically materialized.

**Review by salkinium — APPROVED** [review]
Yes please

**Comment by salkinium** [comment]
I fixed the commit authors and will merge after the CI passes

**Review by chris-durand — APPROVED** [review]

**Pull request merged**